### PR TITLE
drivers/mma8x5x: remodeled driver

### PR DESCRIPTION
**Pull request by haukepetersen**
- renamed to mma8x5x to reflect the versatility
- cleaned up the interface
- fixed value computation for non 12-bit devices of the family
- added default configuration with mma8x5x_params.h

### CODE REVIEW

**Comment by haukepetersen** [comment]
Ups, noticed #5433 just now, so I guess this PR reflects my opinion on that topic :-)

**Comment by haukepetersen** [comment]
There was still one more bug in the value computation, now it should be all good.

**Comment by PeterKietzmann** [comment]
@vincent-d I've seen your thumbs-up reaction to this PR. Any capacities for a review/test from your side here :-)?

**Comment by vincent-d** [comment]
@PeterKietzmann I'll take a look

**Comment by haukepetersen** [comment]
thanks!

File: drivers/mma8x5x/mma8x5x.c [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+#include <stdint.h>
+#include <stdbool.h>
+#include <string.h>
+
+#include "assert.h"
+#include "periph/i2c.h"
+#include "mma8x5x.h"
+#include "mma8x5x_regs.h"
+
+#define ENABLE_DEBUG        (1)
```
Comment (vincent-d) [inline drivers/mma8x5x/mma8x5x.c]: To be disabled

**Comment by vincent-d** [comment]
Appart from a few comments it's seems OK. I quickly tested it on our hardware, it seems to work fine with a mma8652. I must have a board with a mma8653 somewhere, if I can find it I'll try it tomorrow.

File: drivers/include/mma8x5x.h [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+    uint8_t rate;               /**< sampling rate to use */
+    uint8_t range;              /**< scale range to use */
+    uint8_t offset[3];          /**< data offset in X, Y, and Z direction */
+} mma8x5x_params_t;
+
+/**
+ * @brief   Device descriptor for MMA8x5x accelerometers
+ */
+typedef struct {
+    mma8x5x_params_t params;    /**< device configuration parameters */
```
Comment (haukepetersen) [inline drivers/include/mma8x5x.h]: No, this is done intentional, as copying the fields save one de-referencing step every time we interact with the device. Typical memory vs. runtime optimization compromise...

File: drivers/include/mma8x5x.h [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+    int16_t scale;              /**< each count corresponds to (1/scale) g */
```
Comment (haukepetersen) [inline drivers/include/mma8x5x.h]: true.

File: drivers/mma8x5x/mma8x5x.c (hunk 1 above)
Comment (haukepetersen) [inline drivers/mma8x5x/mma8x5x.c]: yupp

**Comment by haukepetersen** [comment]
should be all fixed.

File: drivers/mma8x5x/include/mma8x5x_params.h [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+                                  .addr   = MMA8X5X_PARAM_ADDR, \
+                                  .type   = MMA8X5X_PARAM_TYPE, \
+                                  .rate   = MMA8X5X_PARAM_RATE, \
+                                  .range  = MMA8X5X_PARAM_RANGE, \
+                                  .offset = MMA8X5X_PARAM_OFFSET }
+#endif
+/**@}*/
+
+/**
+ * @brief   AT86RF231 configuration
```
Comment (vincent-d) [inline drivers/mma8x5x/include/mma8x5x_params.h]: bad copy paste

**Comment by haukepetersen** [comment]
Fixed copy-paste error and fixed one more issue with a missing include. Hope all should be well now.

File: tests/driver_mma8x5x/main.c [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+int main(void)
+{
+    mma8x5x_data_t data;
+
+    puts("MMA8652 accelerometer driver test application\n");
+    printf("Initializing MMA8652 accelerometer at I2C_DEV(%i)... ",
+           mma8x5x_params->i2c);
+
+    if (mma8x5x_init(&dev, mma8x5x_params) == MMA8X5X_OK) {
```
Comment (cgundogan) [inline tests/driver_mma8x5x/main.c]: @haukepetersen I am a little bit confused: When building for the `pba` board, will `mma8x5x_params` be the one defined in https://github.com/RIOT-OS/RIOT/pull/6153/files#diff-0c2546bb5afdcd4356dca7827e018afbR34 or the one defined in https://github.com/RIOT-OS/RIOT/pull/6153/files#diff-e3371ca688df37fc060420df8b71ac3bR66 ?

Comment (haukepetersen) [inline tests/driver_mma8x5x/main.c]: the params file in the board (or even one in the app folder) always 'overrides' the default one in the drivers include folder. See also https://github.com/RIOT-OS/RIOT/wiki/Guide:-Writing-a-device-driver-in-RIOT

**Comment by PeterKietzmann** [comment]
@vincent-d just one click away. Would you ACK this  PR?

**Comment by vincent-d** [comment]
@PeterKietzmann @haukepetersen ACK !

**Pull request merged**